akka-http-tests/src/test/scala/akka/http/scaladsl/server/AkkaHttpServerThroughputSpec.scala

```diff
@@ -0,0 +1,225 @@
+          }
+      }
+  }
+
+  def infoThe(lines: Array[String]): Unit =
+    lines.foreach(l ⇒ info("  " + l))
+
+  def printWrkPercentiles(prefix: String, lines: Array[String]): Unit = {
+    val percentilesToPrint = 8
+
+    def durationAsMs(d: String): Long = {
+      val dd = d.replace("us", "µs") // Scala Duration does not parse "us"
+      Duration(dd).toMillis
+    }
+
+    var i = 0
+    val correctedDistributionStartsHere = lines.zipWithIndex.find(p ⇒ p._1 contains "Latency Distribution").map(_._2).get
+
+    i = correctedDistributionStartsHere + 1 // skip header
+    while (i < correctedDistributionStartsHere + 1 + percentilesToPrint) {
+      val line = lines(i).trim
+      val percentile = line.takeWhile(_ != '%')
+      println(prefix + percentile + "_corrected," + durationAsMs(line.drop(percentile.length + 1).trim))
+      i += 1
+    }
+
+    val uncorrectedDistributionStartsHere = lines.zipWithIndex.find(p ⇒ p._1 contains "Uncorrected Latency").map(_._2).get
+
+    i = uncorrectedDistributionStartsHere + 1 // skip header
+    while (i < uncorrectedDistributionStartsHere + 1 + percentilesToPrint) {
+      val line = lines(i).trim
+      val percentile = line.takeWhile(_ != '%')
+      println(prefix + percentile + "_uncorrected," + durationAsMs(line.drop(percentile.length + 1).trim))
+      i += 1
+    }
+  }
+
+  def printAbPercentiles(prefix: String, lines: Array[String]): Unit = {
+    val percentilesToPrint = 9
+
+    def durationAsMs(d: String): Long =
+      Duration(d).toMillis
+
+    var i = 0
+    val correctedDistributionStartsHere = lines.zipWithIndex.find(p ⇒ p._1 contains "Percentage of the requests").map(_._2).get
+
+    i = correctedDistributionStartsHere + 1 // skip header
+    while (i < correctedDistributionStartsHere + 1 + percentilesToPrint) {
+      val line = lines(i).trim
+      val percentile = line.takeWhile(_ != '%')
+      println(prefix + percentile + "," + durationAsMs(line.drop(percentile.length + 1).replace("(longest request)", "").trim + "ms"))
+      i += 1
+    }
+  }
+
+  var _ifWrk2Available: Option[Boolean] = None
+  @tailrec final def ifWrk2Available(test: ⇒ Unit): Unit = {
+    _ifWrk2Available match {
+      case Some(false) ⇒ throw new TestPendingException()
+      case Some(true)  ⇒ test
+      case None ⇒
+        import scala.sys.process._
+
+        val wrk = """wrk""".!
+        //        def wrk2 = """wrk2""".!
+        _ifWrk2Available = Some(wrk == 1) // app found, help displayed
+        ifWrk2Available(test)
+    }
+  }
+
+  var _ifAbAvailable: Option[Boolean] = None
+  @tailrec final def ifAbAvailable(test: ⇒ Unit): Unit = {
+    _ifAbAvailable match {
+      case Some(false) ⇒ throw new TestPendingException()
+      case Some(true)  ⇒ test
+      case None ⇒
+        import scala.sys.process._
+
+        val wrk = """ab -h""".!
+        _ifAbAvailable = Some(wrk == 22) // app found, help displayed
+        ifAbAvailable(test)
+    }
+  }
```

Comment:
yeah 22 is a weird return code `ab` gives when it prints help... I'll comment on it